compiler-core/src/language_server/engine.rs

```diff
@@ -417,6 +415,44 @@ where
 
         completions
     }
+
+    pub fn inlay_hint(&mut self, params: lsp::InlayHintParams) -> Response<Option<Vec<InlayHint>>> {
```

Comment:
What's the difference between `Ok(None)` and `Ok(Some(vec![]))`?

It seems they are semantically the same, so if that's the case let's remove the option as the engine class is intended to be a simpler API rather than a copy of the protocol.